crawl-ref/source/player-stats.cc

```diff
@@ -690,3 +691,132 @@ bool have_stat_zero()
 
     return false;
 }
+
+/*Gnoll ability to shift three points from one attribute to another,
+ *becoming drained if successful.
+ *
+ *Cannot shift below stat zero, and cannot use if drained (checked in
+ *ability.cc)
+*/
+spret_type gnoll_shift_attributes()
+{
+    stat_type source_attribute;
+    stat_type destination_attribute;
+    int keyin;
+#ifdef TOUCH_UI
```

Comment:
The TOUCH_UI stuff doesn't look right to me. I don't think this will even compile with `TOUCH_UI` defined, because you re-declare the same variable in both prompts. If it does compile it's a memory leak because you `new` up all these pointers and never `delete` them.

Reply:
Fixed by moving TOUCH_UI and switch statements for choosing a stat into a single function instead.